src/NameRegistry.sol

```diff
@@ -909,6 +919,39 @@ contract NameRegistry is
         }
     }
 
+    /**
+     * @notice Move the fnames froms the current owners to desired destinations and renew them for another week.
+     *         Does not work when paused because it calls _transfer.
+     *
+     * @param reclaimActions an array of ReclaimAction structs representing the fnames and their corresponding
+     *           destination addresses.
+     */
+    function reclaim(ReclaimAction[] calldata reclaimActions) external payable {
```

Comment:
it would be nice to remove the basic reclaim function as long as the gas costs isnt much higher when calling this with a single item. could you calculate the diff between reclaim with an array of one reclaim action and calling reclaim directly?

Reply:
Of course!

Reply:
So I wrote two basic test cases are identical to each other except for the last line.

```
    function testOldReclaim(address mod, address alice) public {
        _assumeClean(mod);
        _assumeClean(alice);
        _register(alice);
        _grant(MODERATOR_ROLE, mod);
        NameRegistry.ReclaimAction[] memory reclaimActions = new NameRegistry.ReclaimAction[](1);
        reclaimActions[0] = NameRegistry.ReclaimAction(ALICE_TOKEN_ID, POOL);
        vm.prank(mod);
        nameRegistry.reclaim(ALICE_TOKEN_ID);
    }
```

```
    function testNewReclaim(address mod, address alice) public {
        _assumeClean(mod);
        _assumeClean(alice);
        _register(alice);
        _grant(MODERATOR_ROLE, mod);
        NameRegistry.ReclaimAction[] memory reclaimActions = new NameRegistry.ReclaimAction[](1);
        reclaimActions[0] = NameRegistry.ReclaimAction(ALICE_TOKEN_ID, POOL);
        vm.prank(mod);
        nameRegistry.reclaim(reclaimActions);
    }
```

The .gas-snapshot reported the following:
```
NameRegistryTest:testNewReclaim(address,address) (runs: 512, μ: 204044, ~: 204044)
NameRegistryTest:testOldReclaim(address,address) (runs: 512, μ: 205306, ~: 205306)
```

For this test, I used the `unchecked` pattern as you suggested.

Reply:
@varunsrin if this looks good, I can go ahead and replace the current reclaim function with the proposed one.

Reply:
its surprising that the new reclaim function is cheaper than the old one. it has one more assignment `uint256 reclaimActionsLength = reclaimActions.length;` so i would expect it to take a bit more gas. 

perhaps its something with the calldata param saving gas? or could there be something with the setup of the test?

if you are confident there is nothing wrong with the test, go ahead and replace it. i'll do one final test pass before approving, just to make sure that we're not missing anything.

Reply:
Yes, it is surprising!

To further check on this, I went ahead and changed the proposed function's 8 different test cases to only perform reclaim for a single fname.

I put the changes [here](https://github.com/gaavar/contracts/blob/feat-reclaim-multiple-gas-tests/test/NameRegistry.t.sol) in a separate branch on my fork for testing/review.

The `forge test --gas-report` gave similar results.

```
| reclaim((uint256,address)[])               | 2091            | 17000  | 24949  | 26012  | 8       |
| reclaim(uint256)                           | 2513            | 17860  | 26164  | 27226  | 8       |
```

Based on this, I am going to go ahead and replace the current reclaim.

Reply:
👍